src/Course2/week1/StronglyConnectedComponents.java

```diff
@@ -0,0 +1,82 @@
+package Course2.week1;
+
+import java.util.*;
+
+public class StronglyConnectedComponents {
+    private void dfs(int node, Stack<Integer> st, ArrayList<ArrayList<Integer>> adj, int vis[]) {
+        vis[node] = 1;
+        for(Integer it : adj.get(node)) {
+            if(vis[it] == 0) {
+                dfs(it, st, adj, vis);
+            }
+        }
+
+        st.push(node);
+    }
+
+    private void revDfs(int node, ArrayList<ArrayList<Integer>> transpose, int vis[]) {
+        vis[node] = 1;
+        System.out.print(node + " ");
+        for(Integer it : transpose.get(node)) {
+            if(vis[it] == 0) {
+                revDfs(it, transpose, vis);
+            }
+        }
+    }
+
+    void kosaRaju(ArrayList<ArrayList<Integer>> adj, int n)
+    {
+        int vis[] = new int[n];
+        Stack<Integer> st = new Stack<Integer>();
+        for(int i = 0;i<n;i++) {
+            if(vis[i] == 0) {
+                dfs(i, st, adj, vis);
+            }
+        }
+
+        ArrayList<ArrayList<Integer> > transpose = new ArrayList<ArrayList<Integer> >();
+
+        for (int i = 0; i < n; i++)
+            transpose.add(new ArrayList<Integer>());
+
+        for(int i = 0;i<n;i++) {
+            vis[i] = 0;
+            for(Integer it: adj.get(i)) {
+                transpose.get(it).add(i);
+            }
+        }
+
+        while(st.size() > 0) {
+            int node = st.peek();
+            st.pop();
+            if(vis[node] == 0) {
+                System.out.print("SCC: ");
+                revDfs(node, transpose, vis);
+                System.out.println();
+            }
+        }
+
+    }
+    public static void main(String args[])
+    {
+        int n = 7;
+        ArrayList<ArrayList<Integer> > adj = new ArrayList<ArrayList<Integer> >();
+
+        for (int i = 0; i < n; i++)
```

Comment:
use array instead of arraylist. too much initialization happening in code. Think of a better approach